builder/simplify.cc

```diff
@@ -729,7 +729,7 @@ class simplifier : public node_mutator {
           }
         } else if (op->intrinsic == intrinsic::buffer_at) {
           for (int d = 0; d < static_cast<int>(std::min(info->dims.size(), args.size() - 1)); ++d) {
-            if (prove_true(args[d + 1] == info->dims[d].bounds.min)) {
+            if (prove_true(args[d + 1] == info->dims[d].bounds.min) && !info->dims[d].fold_factor.defined()) {
               // This argument is equal to the default value, and we know it is in bounds.
```

Comment:
There should probably be a similar simplification when the fold factor is defined, except comparing to 0 instead of the min.

Reply:
What would you set args[d+1] to though? expr() would be incorrect because it implies that the coordinate is min which is incorrect for folded buffers.

Reply:
If the argument is 0, and the sim is folded, then 0 is the default value, so it could be replaced by expr().

In other words, this could be rewritten to compute a default value (0 if folded, buffer_min otherwise) and then use that in the original simplification check

Reply:
Done.